Islandora/islandora
# Some deprecations in advance of Drupal 11

**GitHub Issue**: https://github.com/Islandora/documentation/issues/2318

* Other Relevant Links (Google Groups discussion, related pull requests,
 Release pull requests, etc.)

# What does this Pull Request do?
Removes deprecated code per the following:

* [New API for defining Field Type categories](https://www.drupal.org/node/3375748)
* [EntityReferenceTestTrait has been renamed to EntityReferenceFieldCreationTrait](https://www.drupal.org/node/3401941)
* and updating the cache action version in CI.

# How should this be tested?

* [Run tests locally](https://islandora.github.io/documentation/technical-documentation/running-automated-tests/) (deprecation notices seem to be suppressed in CI). You shouldn't be able to track down these errors to issues in Islandora code (note that not all contrib modules have used the new API for defining field type categories so those may pop up but it seems it's not our fault.)

# Documentation Status

* Does this change existing behaviour that's currently documented? no!
* Does this change require new pages or sections of documentation? no!
* Who does this need to be documented for? nobody!
* Associated documentation pull request(s): ___  or documentation issue ___

# Additional Notes:
Any additional information that you think would be helpful when reviewing this
 PR.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/committers

Draft because there are a lot more problems when running tests locally.
